fc/parsing/actions.py

```diff
@@ -521,47 +523,101 @@ def _expr(self):
         return self
 
 
-class InputVariable(BaseGroupAction):
-    """
-    Parse action for input variables defined in the model interface.
+class VariableReference:
+    """Mixin providing properties for resolving variable references.
+
+    Used by input & output variable specifications, inter alia.
+
+    Properties:
+
+    ``prefixed_name``
+        A 'prefix:local_name' string.
+
+    ``ns_prefix``
+        The namespace prefix part of ``prefixed_name``.
+
+    ``local_name``
+        The local name part of ``prefixed_name``.
+
+    ``ns_uri``
+        Once namespace prefixes have been resolved, the namespace URI corresponding to ``ns_prefix``.
+
+    ``rdf_term``
+        The RDF term that annotates variable(s) we reference.
 
-    ``input <prefix:term> [units <uname>] [= <initial_value>]``
     """
     def _expr(self):
         name = self.prefixed_name = self.get_named_token_as_string('name')
         self.ns_prefix, self.local_name = name.split(':', 1)
         self.ns_uri = None  # Will be set later using protocol's namespace mapping
+        return self
+
+    @property
+    def rdf_term(self):
+        """The RDF term annotating this variable."""
+        return create_rdf_node((self.ns_uri, self.local_name))
+
+    @classmethod
```

Comment:
Any reason why this method can't live in the test code?

Reply:
Because the test code then needs to know details about exactly what properties to set on this class, which seemed ugly to me.